jdaviz/configs/imviz/plugins/viewers.py

```diff
@@ -232,14 +235,20 @@ def _get_real_xy(self, image, x, y, reverse=False):
         in Subset Tools plugin). Never use this for coordinates display panel.
 
         """
+        within_bounding_box = True
         if data_has_valid_wcs(image):
             # Convert these to a SkyCoord via WCS - note that for other datasets
             # we aren't actually guaranteed to get a SkyCoord out, just for images
             # with valid celestial WCS
             try:
+                link_type = self.get_link_type(image.label)
+                if link_type in ('wcs', 'self'):
+                    within_bounding_box = wcs_utils.data_within_gwcs_bounding_box(
```

Comment:
I moved the code out to its own function in `wcs_utils.py` in hope that it might come in handy if you want to distinguish between different kinds of out-of-bounds situations.